### PR TITLE
weather app-Ivett

### PR DESCRIPTION
**Pull request by Lilivet**
This is the final project for the weather app

### CODE REVIEW

**Comment by Lilivet** [comment]
weather-app-ivett.netlify.com

**Comment by ericaglimsholt** [comment]
Hi Ivett! Your project still doesn't work. Have you had some time to look at it?

**Comment by Lilivet** [comment]
Hi Erica,

It needs to allow javascript and then it works.

Have you tried to allow java script? I do not know what to remove this
step. I will ask Demien and Jenny.

Regards,
Ivett

**Comment by Lilivet** [comment]
I found out why it was failing. I had in my code http and not https on the
url for the API calls.

Now it is working fine!!

Regards,
Ivett

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="style.css">
+    <title>Document</title>
+</head>
+
+<body style="" id="background">
```
Comment (ericaglimsholt) [inline code/index.html]: Why is there an empy style prop here? :)

File: code/index.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <center>
+        <section class="flex-parent-main">
+            <h2>Today´s weather in:</h2>
+            <header>
+                <h1 class="city" id="city-name"></h1>
+                <div class="icon" id="icon"></div>
+                <div class="parentDegrees">
+                    <h1 class="degrees" id="temp-degrees"></h1>
+                    <h1 class="sky-weather" id="sky-display"></h1>
```
Comment (ericaglimsholt) [inline code/index.html]: Could suit as a `<h2>` or `<h3>`. `<h1>` should describe the page

File: code/javascript.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
+//fetch weather for Stockholm
+fetch("https://api.openweathermap.org/data/2.5/weather?q=Stockholm,Sweden&units=metric&APPID=81056f8102cf170efbadf6ea579c361c")
+    .then((response) => {
+        return response.json()
+    }).then((json) => {
+        console.log(json)
```
Comment (ericaglimsholt) [inline code/javascript.js]:
```suggestion

```

File: code/javascript.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        const jsonText = JSON.stringify(json);
+        setHtmlData(json);
+    }).catch((err) => {
+        console.log("caught error", err)
+    });
+
+//Function to round the degrees
+function round(degrees, precision) {
+    var multiplier = Math.pow(10, precision || 0);
```
Comment (ericaglimsholt) [inline code/javascript.js]:
```suggestion
    let multiplier = Math.pow(10, precision || 0);
```

File: code/javascript.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    // degrees & state of the sky
+    const degrees = data.main.temp;
+    document.getElementById("temp-degrees").innerText = round(degrees) + " ° ";
+    const stateOfTheSky = data.weather[0].main;
+    document.getElementById("sky-display").innerText = stateOfTheSky;
+    const iconOfState = data.weather[0].icon;
+    document.getElementById("icon").innerHTML = `<img src="https://openweathermap.org/img/wn/${iconOfState}@2x.png" alt="" />`;
+
+    // defining background color 
+    var color;
```
Comment (ericaglimsholt) [inline code/javascript.js]:
```suggestion
    let color;
```

File: code/javascript.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    const sunsetTime = sunset.toLocaleTimeString([], { timeStyle: 'short' })
+    document.getElementById("sunset").innerText = "Sunset: " + sunsetTime;
+    document.getElementById("sunrise").innerText = "Sunrise: " + sunriseTime;
+}
+//fetch forecast for 7 days of the week in Stockholm 
+fetch("https://api.openweathermap.org/data/2.5/forecast?q=Stockholm,Sweden&units=metric&APPID=81056f8102cf170efbadf6ea579c361c")
+    .then((response) => {
+        return response.json()
+    }).then((response2) => {
+        console.log(response2)
```
Comment (ericaglimsholt) [inline code/javascript.js]:
```suggestion

```

File: code/javascript.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        console.log("caught error", err1)
+    });
+
+const weekDays = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+
+function daysOfWeekData(json) {
+    //days of the week and temperature
+    for (var i = 0; i < 40; i += 8) {
+        let theDate = new Date(json.list[i].dt * 1000)
+        console.log(weekDays[theDate.getDay()] + " " + json.list[i].main.temp);
```
Comment (ericaglimsholt) [inline code/javascript.js]:
```suggestion

```

File: code/javascript.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        console.log("day" + ((i / 8) + 1));
```
Comment (ericaglimsholt) [inline code/javascript.js]:
```suggestion

```

File: code/style.css [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    border-bottom: 1px solid grey;
+}
+
+@media (min-width: 400px) {
+.flex-parent-main {
+    width: 400px;
+}
+.forecast {
+    width: 400px;
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion

```

File: code/style.css [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+}
+th {
+    /* border: 3px solid grey; */
+    width: 40%;
+    height: 6px;
+    padding-top: 15px;
+    border-bottom: 1px solid grey;
+}
+td {
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion

```

File: code/style.css [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    width: 100%;
+
+}
+tr {
+    font-size: 25px;
+   
+}
+th {
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion

```

File: code/style.css [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.table {
+    /* border: 3px solid grey; */
+    width: 100%;
+
+}
+tr {
+    font-size: 25px;
+
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion
```

File: code/style.css [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+}
+.sun-container {
+    font-size: 25px;
+    padding: 30px;
+}
+
+.table {
+    /* border: 3px solid grey; */
+    width: 100%;
+
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion
```

File: code/style.css [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    /* border: 3px solid grey; */
+    padding: 8px; 
+}
+.sun-container {
+    font-size: 25px;
+    padding: 30px;
+}
+
+.table {
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion
```

File: code/style.css [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+    justify-content: center;
+}
+.degrees {
+    flex-wrap: wrap;
+    /* border: 3px solid grey; */
+    padding: 8px;
+}
+.sky-weather {
+    flex-wrap: wrap;
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion
```

File: code/style.css [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.parentDegrees {
+    /* border: 3px solid grey; */
+    display: flex;
+    justify-content: center;
+}
+.degrees {
+    flex-wrap: wrap;
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion
```

File: code/style.css [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+h1 {
+   font-size: 40px;
+}
+
+h2 {
+   font-size: 30px;
+}
+
+.parentDegrees {
+    /* border: 3px solid grey; */
```
Comment (ericaglimsholt) [inline code/style.css]:
```suggestion
```

**Pull request closed without merging**